sphinx/ext/autosummary/__init__.py

```diff
@@ -502,6 +503,20 @@ def import_by_name(name, prefixes=[None]):
     raise ImportError('no module named %s' % ' or '.join(tried))
 
 
+def _import_or_importerror(modname):
+    """
+    Call __import__(modname), convert exceptions to ImportError
+    """
+    try:
+        return __import__(modname)
```

Comment:
We also have similar process on `sphinx.ext.autodoc`. There are additional process for handling warnings. It would be nice if integrated.
```
            with warnings.catch_warnings():
                warnings.filterwarnings("ignore", category=ImportWarning)
                with logging.skip_warningiserror(not self.env.config.autodoc_warningiserror):
                    __import__(self.modname)
```

Reply:
If we do this, please do it in `sphinx.ext.autodoc` or a separate utils module: we don't want `sphinx.ext.autodoc` depending on `sphinx.ext.autosummary`